release.sh

```diff
@@ -1,67 +1,59 @@
 #!/bin/bash
 
-APP="app_gtQEt1zkGMj"
-PROJECT="yb"
-TOKEN="${RELEASE_TOKEN}"
-RELEASE_KEY="${RELEASE_KEY}"
+set -euo pipefail
 
-if [ -z "${VERSION}" ]; then
-  echo "Extracting version from tag ref ${YB_GIT_BRANCH}"
-  VERSION="$(echo $YB_GIT_BRANCH | sed -e 's|refs/tags/||g')"
-fi
-
-if [ -z "${VERSION}" ]; then
-  echo "No version provided, won't release"
+VERSION="${1:-$(echo "${YB_GIT_BRANCH:-}" | sed -e 's:^refs/tags/::')}"
+if [[ -z "$VERSION" ]]; then
+  echo "release.sh: no version provided" 1>&2
   exit 1
 fi
 
-STABLE_TAGGED="$(echo $VERSION | grep -o '\-stable')"
-
-if [ -z "${CHANNEL}" ]; then
-  echo "Channel not set, will release as unstable"
-  CHANNEL="unstable"
+if $(echo $VERSION | grep -vqo '^v'); then
+    echo "Doesn't start with a \"v\" when it should, not releasing"
+    exit 1
 fi
 
-if [ "${STABLE_TAGGED}" == "-stable" ]; then
-    # Strips stable
-    STRIPPED_VERSION="$(echo "$VERSION" | sed -e 's/-stable.*//g')"
-    echo "Stable version $VERSION becomes $STRIPPED_VERSION, so decided to release as stable"
-    VERSION=$STRIPPED_VERSION
+if $(echo $VERSION | grep -qo '\-[a-z]\+[0-9]*$'); then
+    # Release candidate release
+    CHANNEL="preview"
+else
+    # Stable releases should not have a suffix
     CHANNEL="stable"
 fi
 
-umask 077
-
-cleanup() {
-    rv=$?
-    rm -rf "$tmpkeyfile"
-    exit $rv
-}
-
-set -x
+local_test_release="${TEST_RELEASE:-}"
+
+# YourBase S3 artifacts Release
+# TODO migrate to CATS
+AWS_ACCESS_KEY_ID="${AWS_ACCESS_KEY_ID:-}"
+AWS_SECRET_ACCESS_KEY="${AWS_SECRET_ACCESS_KEY:-}"
+aws_disabled=true
+[ -n "${AWS_ACCESS_KEY_ID}" -a -n "${AWS_SECRET_ACCESS_KEY}" ] && aws_disabled=false
+if $aws_disabled; then
+    echo "No AWS credentials, probably in Staging/Preview, giving up"
+    # Non fatal, we won't get GH status errors when trying to release on Staging/Preview
+    exit 0
+fi
 
-tmpkeyfile="$(mktemp)"
-trap "cleanup" INT TERM EXIT
+# Commit info
+COMMIT="${YB_GIT_COMMIT:-}"
+if [ -z "${COMMIT}" ]; then
+    # If git is installed
+    if hash git; then
+        COMMIT="$(git rev-list -n1 HEAD)"
```

Comment:
Slightly more compact way to write:

```suggestion
        COMMIT="$(git rev-parse HEAD)"
```